provider/ecs.go

```diff
@@ -206,13 +206,30 @@ func (provider *ECS) listInstances(ctx context.Context, client *awsClient) ([]ec
 		taskArns = append(taskArns, req.Data.(*ecs.ListTasksOutput).TaskArns...)
 	}
 
-	req, taskResp := client.ecs.DescribeTasksRequest(&ecs.DescribeTasksInput{
-		Tasks:   taskArns,
-		Cluster: &provider.Cluster,
-	})
+	var splitTaskReq [][]*string
+	for i := 0; i < len(taskArns); i += 100 {
+		sliceEnd := -1
+		if i+100 < len(taskArns) {
+			sliceEnd = i + 100
+		} else {
+			sliceEnd = len(taskArns)
+		}
+		split_taskreq = append(splitTaskReq, taskArns[i:sliceEnd])
+	}
```

Comment:
Some tests on the newly created function would also be nice to cover the representative cases (e.g., 0, 1, 99, 100, 101, 200, 400).